akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/StreamTcpDocTest.java

```diff
@@ -49,46 +50,53 @@ public static void tearDown() {
   }
 
   final FlowMaterializer mat = ActorFlowMaterializer.create(system);
-  
-  final InetSocketAddress localhost = new InetSocketAddress("127.0.0.1", 8889);
-  
+
   private final ConcurrentLinkedQueue<String> input = new ConcurrentLinkedQueue<String>();
   {
     input.add("Hello world");
     input.add("What a lovely day");
   }
-  
+
   private String readLine(String prompt) {
     String s = input.poll();
     return (s == null ? "q": s);
   }
-  
+
   @Test
   public void demonstrateSimpleServerConnection() {
-    //#echo-server-simple-bind
-    final InetSocketAddress localhost = 
-        new InetSocketAddress("127.0.0.1", 8889);
-    // IncomingConnection and ServerBinding imported from StreamTcp
-    final Source<IncomingConnection,Future<ServerBinding>> connections =
+    {
+      //#echo-server-simple-bind
+      final InetSocketAddress localhost = new InetSocketAddress("127.0.0.1", 8889);
+      //#echo-server-simple-bind
+    }
+    {
+      final InetSocketAddress localhost = SocketUtils.temporaryServerAddress();
+      //#echo-server-simple-bind
+      // IncomingConnection and ServerBinding imported from StreamTcp
+      final Source<IncomingConnection, Future<ServerBinding>> connections =
         StreamTcp.get(system).bind(localhost);
-    //#echo-server-simple-bind
+      //#echo-server-simple-bind
 
-    //#echo-server-simple-handle
-    connections.runForeach(connection -> {
-      System.out.println("New connection from: " + connection.remoteAddress());
+      //#echo-server-simple-handle
+      connections.runForeach(connection -> {
+        System.out.println("New connection from: " + connection.remoteAddress());
```

Comment:
Add a TODO around all sysouts here please: we can silence those nicely once my cookbook and this tools is merged: https://github.com/akka/akka/pull/16850/files#diff-0e6ca0634b5f28aa2803caf0ea8ca9c6R125